src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,155 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int ELEMENT_NOT_FOUND_INDEX = -1;
+    private static final int EMPTY_LIST_SIZE = 0;
+    private static final int CHANGE_BY_ONE = 1;
+    private static final int ARRAY_BEGINNING_INDEX = 0;
+    private static final int DEFAULT_ARRAY_GROW_COEFFICIENT = 2;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= elements.length) {
+            defaultGrowArrayIfFull();
+        }
+        elements[size] = value;
+        increaseSize(CHANGE_BY_ONE);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || checkForNegativeIndex(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Cannot add the value by index: "
+                    + "Index is out of bounds");
+        }
+        if (size + CHANGE_BY_ONE > elements.length) {
+            defaultGrowArrayIfFull();
+        }
+        if (index == size) {
+            elements[index] = value;
+        }
+        int numberOfElementsAfterIndex = size - index;
+        copyElements(elements, index, elements, index + CHANGE_BY_ONE, numberOfElementsAfterIndex);
+        elements[index] = value;
+        increaseSize(CHANGE_BY_ONE);
```

Comment:
No need to overdesign. There is nothing more readable than basic commands :)
```suggestion
        size++;
```